server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsService.java

```diff
@@ -654,12 +677,10 @@ public void clusterChanged(ClusterChangedEvent event) {
         } else {
             cancelPollingClusterFormationInfo();
         }
-        if (clusterService.localNode().isMasterNode() == false) {
-            if (currentMaster == null) {
-                beginPollingRemoteMasterStabilityDiagnostic();
-            } else {
-                cancelPollingRemoteMasterStabilityDiagnostic();
-            }
+        if (currentMaster == null && clusterService.localNode().isMasterNode() == false) {
```

Comment:
So both `cancel...` methods are called every time there's a non-null master right? 

Can we combine and simplify the if statemets to reflect this? 
ie.
```
   if master == null { 
       if isMasterEligible { pollABC } else { pollEFG }
    } else {
       cancelABC
       cancelEFG
    }
```

What do you think?

Reply:
Since I moved the call to `beginPollingRemoteMasterStabilityDiagnostic()` out of the constructor, I was able to check whether a node was master-eligible before calling `beginPollingRemoteMasterStabilityDiagnostic()`. So now we only need to cancel it for non-master-eligible nodes. So I put it back the way it used to be.